### ISSUE TITLE
Windows supported mixlib shellout invocation

### ISSUE THREAD

**Issue by webframp** [JamesMGreene/chef-gocd-agent#1]
For https://github.com/JamesMGreene/chef-gocd-agent/blob/master/attributes/default.rb#L31

seems to result in:

```
       Mixlib::ShellOut::InvalidCommandOption
       --------------------------------------
       You must supply both a username and password when supplying a user in windows
```

perhaps that could be adjusted. This is pretty ugly, but maybe it would be enough?

``` ruby
if !(ENV['JAVA_HOME'].nil? || ENV['JAVA_HOME'].empty?)
  default['gocd_agent']['java_home'] = ENV['JAVA_HOME']
else
  case node['platform_family']
  when 'windows'
    cmd = Mixlib::ShellOut.new('echo "${JAVA_HOME}"')
  else
    cmd = Mixlib::ShellOut.new('echo "${JAVA_HOME}"', :user => 'root')
  end
```

curious why the shellout was needed for echo instead of accessing the env var directly via `ENV['JAVA_HOME']`

**Comment by JamesMGreene** [comment]
> curious why the shellout was needed for echo instead of accessing the env var directly via ENV['JAVA_HOME']

`ENV` only contains environment variable values that were set when the Ruby environment was created at the start of a chef-client run.  In other words, it will only contain values that were pre-existing... and that the system even had Java installed, for that matter!

One of the things this cookbook does is install Java as a dependency using the `java` cookbook, which may create/update the `JAVA_HOME` environment variable value during the convergence phase of the chef-client run.

Anyway, the extra `echo`s were my feeble attempt to get an updated value for `JAVA_HOME` but half the time it doesn't really seem to work since UNIX/Linux doesn't exactly have the same concept of a global/system-wide environment variable like Windows does.

**Comment by JamesMGreene** [comment]
I [hopefully?] resolved this in e1934beeaffb6014ebeae04fc9035fa81a397fd9. Released as version `1.1.11`.

Please let me know if you still run into trouble with it.  Thanks!
